### PR TITLE
[Feature #158531993] Add tests for api endpoints.

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
#### What does this PR do?
Creates tests for various edge cases for the api endpoints
#### Description of Task to be completed?
Have tests for the following endpoints `failing`
```
- GET /rides  Fetch all ride offers  
- GET /rides/<rideId>  Fetch a single ride offer 
- POST /rides  Create a ride offer 
- POST /rides/<rideId>/requests  
```
#### How should this be manually tested?
using your terminal,
- clone this repo
- `cd ride-my-way`
- set and activate your virtual environment
- `git checkout ft-api-tests-158531993`
- Run `nosetests`

#### Any background context you want to provide?
Unit testing gives peace of mind as most of the bugs are identified and corrected at early stages of product development.
#### What are the relevant pivotal tracker stories?
[[#158531993]](https://www.pivotaltracker.com/story/show/158531993)

#### How to Review
Look at the file named `testrides.py` and let me know whether I have left out any edge case.

### CODE REVIEW

**Comment by TheSteelGuy** [comment]
You have covered  quite something, i will let you know if there is an edge case to add

File: tests/testrides.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        self.assertEqual(response_data['message'],
+                         "make sure you provide all required fields.")
+
+    def test_available_space_can_only_be_numbers(self):
+        ride = {
+            "start point": "Witeithye",
+            "destination": "Ngara",
+            "route": "Thika superhighway",
+            "start time": "June 10 2018 6:00AM",
+            "available space": "10"
```
Comment (JoshuaOndieki) [inline tests/testrides.py]: Since the string **"10"** can be converted to a number, try testing with letters or other characters

File: tests/testrides.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    def test_user_cannot_get_ride_that_does_not_exist(self):
+        """Assumes no ride with a negative id number"""
+        response = self.app.get('/api/v1/rides/-1',
+                                content_type='application/json')
+        self.assertEqual(response.status_code, 404)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'], 'Ride does not exist')
+
+    def test_user_can_join_a_ride(self):
+        """test user can join a ride"""
```
Comment (JoshuaOndieki) [inline tests/testrides.py]: I assume making a request is not actually joining the ride. My suggestion therefore, is to rename 
`test_user_can_join_a_ride` to `test_user_can_request_a_ride` and 
`test_user_cannot_join_a_non_existing_ride` to `test_user_cannot_request_a_non_existing_ride`

File: tests/testrides.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        response = self.app.post('/api/v1/rides/1/requests',
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 200)
```
Comment (JoshuaOndieki) [inline tests/testrides.py]: I think the appropriate status code for creating a resource, I.e. a **ride request** would be 201 instead of 200

Comment (meshack-mbuvi) [reply tests/testrides.py]: corrected sir.Thanks for your feedback

File: tests/testrides.py (hunk 2 above)
Comment (meshack-mbuvi) [reply tests/testrides.py]: This has been taken care of

File: tests/testrides.py (hunk 1 above)
Comment (meshack-mbuvi) [reply tests/testrides.py]: I have modified it to use 'Ten' instead of "10"

File: requirements.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 aniso8601==3.0.2
 click==6.7
```
[the remark this answers is not in the document]
Comment (meshack-mbuvi) [reply requirements.txt]: They were installed by pip when installing flask and flask_restplus. I don't require them myself

File: requirements.txt [hunk 5]
```diff
@@ -1,5 +1,6 @@
...
+coverage==4.5.1
 Flask==1.0.2
 flask-restplus==0.11.0
 gunicorn==19.8.1
```
[the remark this answers is not in the document]
Comment (meshack-mbuvi) [reply requirements.txt]: Which package?

File: .coveragerc [hunk 6]
```diff
@@ -2,4 +2,8 @@
 omit = */python?.?/*
     */site-packages/*
     */usr/*
-    */tests/*
\ No newline at end of file
+    */tests/*
+    */instance/*
+    *__init__.*
+    /*models*/
+show_missing = True
```
[the remark this answers is not in the document]
Comment (meshack-mbuvi) [reply .coveragerc]: Done

File: requirements.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (LWanjiru) [reply requirements.txt]: gunicorn

**Comment by LWanjiru** [comment]
Is it possible to share a screenshot of the failing tests? @meshack-mbuvi

File: requirements.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (meshack-mbuvi) [reply requirements.txt]: I will use gunicorn to run the app on heroku

**Comment by meshack-mbuvi** [comment]
I wrote the tests first on a different branch which I had not merged since I was waiting your approval. The endpoints are now on their respective branches which still have not merged because of the stated reason above. Merging the branches will make the tests to locate the necessary endpoints and am sure they would pass because have tested them on my machine before pushing to github. With your approval, I would wish to merge the branches so we can see what happens.
